### DISCUSSION TITLE
render CheckTreePicker inline?

### DISCUSSION THREAD

**Discussion by ForeshadowRU** [rsuite/rsuite#3688] [Q&A] [answered]
Hi. I need to display ChechTreePicker in other popup contexts like drawers modals and popups. 
The problem is that picker-popup is rendered in the bottom of the DOM so any click on it is considered as click outside of drawer/modal, hence you can only click once then drawer/modal closes.

Is there any workaround?

**Comment by simonguo** [comment]
https://stackblitz.com/edit/react-bjrmtz?file=index.js

How can you reproduce your problem using the above example?

**Reply by simonguo** [reply]
In addition, you can try to set a container property for CheckTreePicker to specify the container of the popup layer to Modal.

**Reply by ForeshadowRU** [reply]
I will try to, though it will take a while.
I respond asap

**Reply by ForeshadowRU** [reply]
@simonguo https://codesandbox.io/p/sandbox/practical-curran-forked-4v7zx7?file=%2Fsrc%2FApp.jsx

stackblitz doesn't quite work with lib I using for sidebar, so I hope codesandbox is good alternative.

**Reply by simonguo** [reply, marked as answer]
Just set up a container for CheckTreePicker and let it render inside Sidebar.

```tsx
const contentRef = React.useRef();

<div ref={contentRef} style={{ position: "relative" }}>
    <CheckTreePicker
      defaultExpandAll
      data={data}
      container={() => contentRef.current}
    />
</div>
```

**Reply by ForeshadowRU** [reply]
Thank you very much
